src/core/ext/transport/chttp2/transport/chttp2_transport.cc

```diff
@@ -1725,31 +1729,32 @@ class GracefulGoaway : public grpc_core::RefCounted<GracefulGoaway> {
 
   static void OnPingAckLocked(void* arg, grpc_error_handle /* error */) {
     auto* self = static_cast<GracefulGoaway*>(arg);
-    grpc_timer_cancel(&self->timer_);
+    if (self->timer_handle_ != TaskHandle::kInvalid) {
```

Comment:
Once you let the callback manage the ref lifetime, I think this whole chunk of code can condense down to just:
```
self->t_->event_engine->Cancel(std::exchange(self->timer_handle_, TaskHandle::kInvalid));
```

Reply:
This is slick! Thanks!